Sources/Build/Command.link(SwiftModule).swift

```diff
@@ -48,14 +48,15 @@ extension Command {
             return Command(node: product.targetName, tool: ArchiveTool(inputs: inputs, outputs: outputs))
         }
 
-        let inputs = objects
-
         switch product.type {
         case .Library(.Static):
             args.append(outpath)
         case .Test:
             args += ["-module-name", product.name]
           #if os(OSX)
+            // Link all the Clang Module's objects too into XCTest executable on OSX since XCTest is available in ObjC.
+            objects += product.modules.flatMap{ $0 as? ClangModule }.flatMap{ ClangModuleBuildMetadata(module: $0, prefix: prefix, otherArgs: []).objects }
```

Comment:
We can error out in package loading for linux but I think this should still be in OSX conditional because this is the only case where we are linking Swift and C objects together, for other modules we interpolate via modulemaps.